### PR TITLE
Update hdfgroup packages to new URL structure

### LINKED ISSUE #4626: hdf5 download paths
It seems that hdf5 download paths have changed, for some of the versions. Is anybody in charge of hdf5, or can anybody submit a PR?

### PR DESCRIPTION
**Pull request by adamjstewart**
Closes #4626 

The HDF Group updated their website with a new repository structure that seems to be significantly more consistent than the old one (see `url_for_version` for `hdf5` package).

Also updates `hdf` to `AutotoolsPackage` and adds newer versions of `szip`, `hdf`, and `hdf5@1.8`.

@iulian787 @cyrush

### CODE REVIEW

**Comment by adamjstewart** [comment]
Another question worth asking is what we want the default version of HDF5 to be. If you look at the [HDF Group downloads page](https://support.hdfgroup.org/downloads/), you'll see that there are separate download links for HDF5 and HDF5-1.10. This leads me to believe that HDF5-1.10 is pretty cutting-edge and might not be as stable or recommended. @citibeth Any thoughts?

**Comment by citibeth** [comment]
Some "funny stuff" happened with HDF5 and NetCDF.  HDF5 upgraded to a new
file format (or something).  NetCDF would build with that version of HDF5,
but not in a way that would be useful or standard; so NetCDF was held back
to older versions of HDF5 for a while.  Later, NetCDF came out with a new
version that was again compatible with the latest HDF5.

Or at least, that's how I remember it.  I wrote it up somewhere in the
Spack GitHub or email list.

 Bottom line is, don't mess with the version of HDF5 used by NetCDF.

**Comment by cyrush** [comment]
thanks @adamjstewart !
